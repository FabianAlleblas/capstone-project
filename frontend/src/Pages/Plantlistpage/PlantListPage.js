import { useHistory } from 'react-router-dom'
import styled from 'styled-components/macro'
import PlusButton from '../../Components/Buttons/PlusButton'
import Header from '../../Components/Header/Header'
import PlantListCard from '../../Components/Plantlistcard/PlantListCard'

export default function PlantListPage({ plantList }) {
  const history = useHistory()

  return (
    <>
      <FixedHeader>My Plants</FixedHeader>
      <ListContainer>
        {plantList.map((plant) => (
          <PlantListCard
            key={plant.id}
            plant={plant}
            onClick={() => openDetailPage(plant.id)}
          />
        ))}
      </ListContainer>
      <FixedPlusButton onClick={openAddForm} />
    </>
  )

  function openAddForm() {
    history.push('/addplant')
  }
  function openDetailPage(id) {
    history.push(`/plant?id=${id}`)
  }
}

const FixedHeader = styled(Header)`
  left: 0;
  position: fixed;
  top: 0;
`

const ListContainer = styled.main`
  display: grid;
  gap: 40px 30px;
  grid-template-columns: 1fr 1fr;
  padding: 100px 30px 90px;
`
const FixedPlusButton = styled(PlusButton)`
  bottom: 20px;
  position: fixed;
  right: 20px;
`
