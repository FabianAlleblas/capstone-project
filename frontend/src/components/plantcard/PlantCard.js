//import styled from 'styled-components/macro'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import monsteraimage from '../../assets/plant-images/3.jpg'

export default function PlantCard({ plantname, plantspecies }) {
  PlantCard.propTypes = {
    plantname: PropTypes.string.isRequired,
    plantspecies: PropTypes.string.isRequired,
  }

  return (
    <CardWrapper>
      <ImgContainer src={monsteraimage} />
      <PlantName>{plantname}</PlantName>
      <PlantSpecies>{plantspecies}</PlantSpecies>
    </CardWrapper>
  )
}

const CardWrapper = styled.div`
  display: grid;
  justify-items: center;
`

const ImgContainer = styled.div`
  background-image: url(${(props) => props.src});
  background-position: center;
  background-size: cover, contain;
  border-radius: 50%;
  height: 100px;
  margin-bottom: 10px;
  width: 100px;
`

const PlantName = styled.p`
  color: var(--primary-plant-font-color);
  font-size: 1.25rem;
`

const PlantSpecies = styled.p`
  color: var(--secondary-plant-font-color);
  font-size: 0.75rem;
`
