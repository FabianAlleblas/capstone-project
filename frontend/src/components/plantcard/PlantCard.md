```js
<PlantCard plantname={'Jack'} plantspecies={'Monstera deliciosa'} />
```
