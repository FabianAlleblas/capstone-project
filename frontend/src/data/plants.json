[
  {
    "id": 1,
    "name": "Carl",
    "species": "Monstera deliciosa"
  },
  {
    "id": 2,
    "name": "Birgitt",
    "species": "Monstera adansonii"
  },
  {
    "id": 3,
    "name": "Karen",
    "species": "Philodendron gloriosum"
  },
  {
    "id": 4,
    "name": "Jim Jones",
    "species": "Rhaphidophora tetrasperma"
  },
  {
    "id": 5,
    "name": "Samuel",
    "species": "Alocasia zebrina"
  },
  {
    "id": 6,
    "name": "Greg",
    "species": "Scindapsus pictus"
  }
]
