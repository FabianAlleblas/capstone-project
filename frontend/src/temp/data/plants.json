[
  {
    "id": 1,
    "name": "Carl",
    "species": "Monstera deliciosa",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 2,
    "name": "Birgitt",
    "species": "Monstera adansonii",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 3,
    "name": "Karen",
    "species": "Philodendron gloriosum",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 4,
    "name": "Jim Jones",
    "species": "Rhaphidophora tetrasperma",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 5,
    "name": "Samuel",
    "species": "Alocasia zebrina",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 6,
    "name": "Greg",
    "species": "Scindapsus pictus",
    "specialInfo": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  }
]
