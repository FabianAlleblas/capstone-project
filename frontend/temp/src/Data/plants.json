[
  {
    "id": 1,
    "name": "Carl",
    "species": "Monstera deliciosa",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 2,
    "name": "Birgitt",
    "species": "Monstera adansonii",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 3,
    "name": "Karen",
    "species": "Philodendron gloriosum",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 4,
    "name": "Jim Jones",
    "species": "Rhaphidophora tetrasperma",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 5,
    "name": "Samuel",
    "species": "Alocasia zebrina",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  },
  {
    "id": 6,
    "name": "Greg",
    "species": "Scindapsus pictus",
    "info": "variegated",
    "daysLeft": 4,
    "weeksLeft": 3
  }
]
