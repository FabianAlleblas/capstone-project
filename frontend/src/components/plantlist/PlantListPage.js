import styled from 'styled-components/macro'
import PlantCard from './PlantCard'

export default function PlantListPage() {
  return (
    <ItemContainer>
      <PlantCard />
    </ItemContainer>
  )
}

const ItemContainer = styled.section``
