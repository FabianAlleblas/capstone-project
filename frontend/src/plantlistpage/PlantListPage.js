import styled from 'styled-components/macro'
import PlantCard from '../components/plantcard/PlantCard'
import plants from '../data/plants.json'

export default function PlantListPage() {
  return (
    <ItemContainer>
      {plants.map(({ id, name, species }) => (
        <PlantCard key={id} plantname={name} plantspecies={species} />
      ))}
    </ItemContainer>
  )
}

const ItemContainer = styled.section`
  display: grid;
  gap: 40px 20px;
  grid-template-columns: 1fr 1fr;
  padding: 20px;
`
