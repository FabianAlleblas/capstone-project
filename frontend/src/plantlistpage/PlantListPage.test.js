import { render } from '@testing-library/react'
import 'jest-styled-components'
import PlantCard from '../components/plantcard/PlantCard'
import PlantListPage from './PlantListPage'
import plants from '../data/plants.json'

describe('PlantListPage', () => {
  it('renders correctly', () => {
    const { container } = render(<PlantListPage />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('displays the plant names and species', () => {
    const { getByText } = render(<PlantListPage />)

    plants.forEach(({ name }) => expect(getByText(name)).toBeInTheDocument())
    plants.forEach(({ species }) =>
      expect(getByText(species)).toBeInTheDocument()
    )
  })
})
