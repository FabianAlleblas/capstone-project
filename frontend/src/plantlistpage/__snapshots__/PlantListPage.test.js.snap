// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlantListPage renders correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c2 {
  background: var(--primary-dark);
  padding: 20px 30px 13px 20px;
  border-radius: 0 0 50px 0;
}

.c3 {
  color: var(--primary-light);
  font-size: 1.75rem;
  font-weight: 700;
  line-height: 1;
}

.c0 {
  position: fixed;
  top: 0;
  left: 0;
}

<div
  class="c0"
>
  <header
    class="c1"
  >
    <div
      class="c2"
    >
      <h1
        class="c3"
      >
        My Plants
      </h1>
    </div>
  </header>
</div>
`;
